hack/print-workspace-status.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This command is used by bazel as the workspace_status_command
+# to implement build stamping with git information. It pulls information
+# from tags named like vX.Y.Z with an optional postfix, e.g., v1.2.3-alpha.
+
+set -euo pipefail
+
+SCRIPT_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+REPO_ROOT=$(cd ${SCRIPT_ROOT}/..; pwd)
```

Comment:
Use `git rev-parse --show-toplevel` instead?

Reply:
Hmm. That doesn't work if the script is being run from outside of the repository. I'm not sure that's a major concern since bazel is running it, and for bazel to run it must be in the workspace.

Thoughts?

Reply:
That's a fair point. I am not hell bent on making this change. I just thought `rev-parse` was more readable and closer to what you were trying to achieve here. But whatever works.

Reply:
I tried this out with bazel just to see, and it works fine, so I'm going to make this simplification. I like having fewer variables, and the `SCRIPT_ROOT` command always feels a bit voodoo-ish to me.